django/db/backends/sqlite3/base.py

```diff
@@ -445,8 +445,8 @@ def _sqlite_datetime_parse(dt, tzname=None, conn_tzname=None):
     return dt
 
 
-def _sqlite_date_trunc(lookup_type, dt):
-    dt = _sqlite_datetime_parse(dt)
+def _sqlite_date_trunc(lookup_type, dt, tzname, conn_tzname):
+    dt = _sqlite_datetime_parse(dt, tzname, conn_tzname)
```

Comment:
I think we should move:
https://github.com/django/django/blob/3d4ffd1ff0eb9343ee41de77caf6ae427b6e873c/django/db/backends/sqlite3/base.py#L433-L444
to a separate hook and call it only if `tzname is not None`. With that we could avoid unnecessary workarounds such as added in `typecast_time()`.

Reply:
Agreed for the workaround in `typecast_time` which is suboptimal, I removed it.
However, I'm not 100% sure what extracting the timezone chunk would bring in this case 🤔 

I changed the way we call it in `django.db.backends.sqlite3.base._sqlite_time_trunc` and I guess it should be fine without the workaround in `typecast_time`.